pennylane/spin/lattice.py

```diff
@@ -35,10 +35,16 @@ class Lattice:
        positions (list[list[float]]): Initial positions of spin cites. Default value is
            ``[[0.0]`` :math:`\times` ``number of dimensions]``.
 
-       boundary_condition (bool or list[bool]): Defines boundary conditions in different lattice axes,
+       boundary_condition (bool | list[bool]): Defines boundary conditions for different lattice axes,
            default is ``False`` indicating open boundary condition.
        neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
            Default is 1 (nearest neighbour).
+       custom_edges (Optional[list(list(tuples))]): Specifies the edges to be added in the lattice.
+           Default value is None, which adds the edges based on neighbour_order.
+           Each element in the list is for a separate edge, and can contain 1 or 2 tuples.
+           First tuple contains the index of the starting and ending vertex of the edge.
+           Second tuple is optional and contains the operator on that edge and coefficient
+           of that operator.
```

Comment:
Should we also state that `neighbour_order` and `custom_edges` cannot be defined together? Like as a note somewhere?